book/6-export.md

```diff
@@ -0,0 +1,168 @@
+
+
+```
+
+To view the results, we can use the `napari` viewer
+to display the Zarr array including the newly written sub-volume.
+
+```{code-cell} ipython3
+np_array_all = zarr_array[:,:,:]
+viewer = napari.Viewer()
+new_layer = viewer.add_image(np_array_all)
+napari.run()
+```
+
++++
+
+## Converting Zarr sub-volumes to NIfTI/DICOM
+
+Next, we'll look at how to convert Zarr sub-volumes to NIfTI and DICOM formats.
+
+To write this sub-volume to a NIfTI file, we can use the `nibabel` library.
+After processing the sub-volume, we can then read it back in as a NIfTI image (`nibabel.nifti1.Nifti1Image` class)
+to get the image data as a NumPy array
+and copy it back in place to the original Zarr array.
+
+```{code-cell} ipython3
+nifti_img = nib.Nifti1Image(np_array, affine=np.eye(4))
+nib.save(nifti_img, 'output.nii')
+
+# Load the NIfTI file
+img = nib.load('output.nii')
+
+# Get the image data as a NumPy array
+zarr_small_nii = img.get_fdata()
+zarr_array[100:200, 100:200, 100:200]=zarr_small_nii
+```
+
+To write this sub-volume to a DICOM file, we can use the `pydicom` library.
+Since DICOM is typically used for 2D slices, we
+will convert the 3D sub-volume to a 2D NumPy array by taking a single slice.
+
+We then need to create a DICOM dataset and populate it with the necessary metadata using `pydicom.dataset`
+where the image data is stored in the `PixelData` attribute.
+
+The `pydicom.dataset.FileDataset` class is then
+saved as a DICOM file.
+
+
+```{code-cell} ipython3
+import numpy as np
+import pydicom
+from pydicom.dataset import Dataset, FileDataset
+import datetime
+
+# Example 2D data (DICOM is usually 2D per file)
+data = np_array[:,:, 0]  # Take the first slice for demonstration
+
+# Create metadata
+file_meta = pydicom.Dataset()
+ds = FileDataset("output.dcm", {}, file_meta=file_meta, preamble=b"\0" * 128)
+
+# Set required values
+ds.PatientName = "Test^Patient"
+ds.PatientID = "123456"
+ds.Modality = "OT"
+ds.StudyInstanceUID = "1.2.3"
+ds.SeriesInstanceUID = "1.2.3.1"
+ds.SOPInstanceUID = "1.2.3.1.1"
+ds.SOPClassUID = pydicom.uid.SecondaryCaptureImageStorage
+ds.file_meta.TransferSyntaxUID = pydicom.uid.ExplicitVRLittleEndian
+
+# Set image data
+ds.Rows, ds.Columns = data.shape
+ds.PhotometricInterpretation = "MONOCHROME2"
+ds.SamplesPerPixel = 1
+ds.BitsAllocated = 16
+ds.BitsStored = 16
+ds.HighBit = 15
+ds.PixelRepresentation = 0
+ds.PixelData = data.tobytes()
+
+# Save to file
+ds.save_as("output.dcm")
+```
+
+After processing the sub-volume, we can then read it back in as a DICOM image to get the image data as a NumPy array
+and copy it back in place to the original Zarr array.
+
+```{code-cell} ipython3
+
+# Load the DICOM file
+ds = pydicom.dcmread('output.dcm')
+
+# Extract the pixel data as a NumPy array
+data = ds.pixel_array
+```
+
+Both the NIfTI and DICOM sub-volumes can be viewed in open-source viewers such as `ITK-SNAP` (see [ITK-SNAP site](https://www.itksnap.org/pmwiki/pmwiki.php)).
```

Comment:
Perhaps mention somewhere that this process is more complex for ome-zarr images that have a series of different resolution levels? When writing the processed image back, you would need to update all resolution levels appropriately, rather than only a single array.